cmd/migratestats/migrate.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	errgo "gopkg.in/errgo.v1"
+	"gopkg.in/juju/charmrepo.v3/csclient/params"
+	"gopkg.in/juju/charmstore.v5/internal/charmstore"
+	"gopkg.in/juju/charmstore.v5/internal/mongodoc"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	statCountersColl = "juju.stat.counters"
+	statTokensColl   = "juju.stat.tokens"
+	migrationsColl   = "migrations"
+)
+
+var keepKinds = []string{
+	"archive-download",
+	"archive-download-promulgated",
+	"archive-delete",
+	"archive-upload",
+}
+
+// squashStats combines entries in the stats counters
+// into 1-hour granularity rather than 1-minute granularity.
+func squashStats(db *mgo.Database) error {
+	seconds := int(charmstore.StatsGranularity / time.Second)
+	var r []struct{}
+	err := db.C(statCountersColl).Pipe([]bson.M{{
+		"$group": bson.M{
```

Comment:
It would be nice if there was a clearer way to represent the query here. Perhaps you could add it as a comment, but then there is no guarantee that it's up to date.